CMakeLists.txt

```diff
@@ -50,6 +50,13 @@ add_definitions(-DQT_NO_KEYWORDS)
 ###########
 ## Build ##
 ###########
+
+# Needed for shared libraries on Windows
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+if(NOT DEFINED BUILD_SHARED_LIBS)
+  option(BUILD_SHARED_LIBS "Build dynamically-linked binaries" ON)
```

Comment:
no where in the ros2 code base does anyone use this approach. Where did you find this? I might consider keeping a checkout of all the files in the ros2.repos file for finding common approaches to these issues. For instance, `https://github.com/ros-perception/laser_geometry/tree/ros2` has dealt with this exact issue in a relatively simple way

Reply:
I had a checkout but none of my search terms popped this up. I think this will solve the dll issue but I don't see anything in the visibility_control.hpp defs that would address the M_PI issue

Reply:
fyi I think I pulled in the dll exporting rules from here
https://github.com/robotology/how-to-export-cpp-library/blob/master/CMakeLists.txt

Reply:
fancy